version_info.go

```diff
@@ -0,0 +1,306 @@
+}
+
+// FileVeLegalTrademarksrsion returns FileVeLegalTrademarksrsion property.
+func (f Info) FileVeLegalTrademarksrsion() string {
+	return f.GetProperty("LegalTrademarks")
+}
+
+// PrivateBuild returns PrivateBuild property.
+func (f Info) PrivateBuild() string {
+	return f.GetProperty("PrivateBuild")
+}
+
+// SpecialBuild returns SpecialBuild property.
+func (f Info) SpecialBuild() string {
+	return f.GetProperty("SpecialBuild")
+}
+
+// New creates Info instance with default locale.
+func New(path string) (Info, error) {
+	info, err := newWithoutLocale(path)
+	if err != nil {
+		return Info{}, xerrors.Errorf("failed to get VersionInfo; %s", err)
+	}
+
+	pageID := info.GetPageID()
+	info.pageID = pageID
+
+	return info, nil
+}
+
+// NewWithLocale creates Info instance with user-defined locale.
+func NewWithLocale(path string, pageID Locale) (Info, error) {
+	info, err := newWithoutLocale(path)
+	if err != nil {
+		return Info{}, xerrors.Errorf("failed to get VersionInfo; %s", err)
+	}
+
+	info.pageID = pageID
+
+	return info, nil
+}
+
+func newWithoutLocale(path string) (Info, error) {
+	pathPtr, err := syscall.UTF16PtrFromString(path)
+	if err != nil {
+		return Info{}, xerrors.Errorf("failed to convert image path to utf16; %s", err)
+	}
+	size, _, err := getFileVersionInfoSizeProc.Call(
+		uintptr(unsafe.Pointer(pathPtr)),
+		0,
+	)
+	if size == 0 {
+		return Info{}, xerrors.Errorf("failed to get memory size for VersionInfo slice; %s", err)
+	}
+	info := make([]byte, size)
+	ret, _, err := getFileVersionInfoProc.Call(
+		uintptr(unsafe.Pointer(pathPtr)),
+		0,
+		uintptr(len(info)),
+		uintptr(unsafe.Pointer(&info[0])),
+	)
+	if ret == 0 {
+		return Info{}, xerrors.Errorf("failed to get VersionInfo from windows; %s", err)
+	}
+
+	vi := Info{data: info}
+
+	fixedInfo := vi.GetFixedInfo()
+	vi.FixedFileInfo = fixedInfo
+
+	return vi, nil
+}
+
+// GetPageID gets pageID filed in structure.
+// It tries to get pageID from VersionInfo data.
+// If getting ends with fail, it returns default pageID
+// 040904E4 // US English + CP_USASCII.
+func (f Info) GetPageID() Locale {
+	data, err := f.verQueryValue(`\VarFileInfo\Translation`, false)
+	if err != nil || len(data) == 0 {
+		return EnglishUSAscii
+	}
+
+	// each pageID consists of a 16-bit language ID and a 16-bit code page
+	type langAndCodePage struct {
+		Language uint16
+		CodePage uint16
+	}
+	if len(data)%int(unsafe.Sizeof(langAndCodePage{})) != 0 {
+		return EnglishUSAscii
+	}
+	n := len(data) / int(unsafe.Sizeof(langAndCodePage{}))
+	if n == 0 {
+		return EnglishUSAscii
+	}
+
+	ids := (*[1 << 28]langAndCodePage)(unsafe.Pointer(&data[0]))[:n:n]
+	return Locale(fmt.Sprintf("%04x%04x", ids[0].Language, ids[0].CodePage))
+}
+
+// GetFixedInfo returns FixedFileInfo structure, which constructs from windows
+// VS_FIXEDFILEINFO structure.
+// source:
+// https://helloacm.com/c-function-to-get-file-version-using-win32-api-ansi-and-unicode-version/
+func (f Info) GetFixedInfo() FixedFileInfo {
```

Comment:
Suppose we have no need to embed `FixedFileInfo` into every `Info` instance if it should be queried. Having a getter should be enough.